File: solutions/python/rna-transcription/1/rna_transcription.py
def to_rna(dna_strand):
    return "".join(map(get_rna, dna_strand))

def get_rna(dna):
    if dna == "G":
        return "C"
    if dna == "C":
        return "G"
    if dna == "T":
        return "A"
    if dna == "A":
        return "U"
    
    raise ValueError("Unknown DNA")

File: solutions/typescript/rna-transcription/1/rna-transcription.ts
function dnaToRnaNucleotide(dna: string): string {
  switch(dna) {
    case 'G': return 'C';
    case 'C': return 'G';
    case 'T': return 'A';
    case 'A': return 'U';
  }
  return '';
}

function isValidDna(dna: string): boolean {
  return 'GCTA'.includes(dna);
}

export function toRna(dna: string): string {
  const dnaNucleotides = dna.split('');

  if (!dnaNucleotides.every(isValidDna)) {
    throw new Error("Invalid input DNA.");
  }
  
  return dnaNucleotides.map(dnaToRnaNucleotide).join('');
}
